src/main/scala/strawman/collection/Iterable.scala

```diff
@@ -302,4 +632,29 @@ trait IterableOps[+A, +CC[X], +C] extends Any {
   /** Zip. Interesting because it requires to align to source collections. */
   def zip[B](xs: IterableOnce[B]): CC[(A @uncheckedVariance, B)] = fromIterable(View.Zip(coll, xs))
   // sound bcs of VarianceNote
+
+  /** Converts this $coll of pairs into two collections of the first and second
+    *  half of each pair.
+    *
+    *    {{{
+    *    val xs = $Coll(
+    *               (1, "one"),
+    *               (2, "two"),
+    *               (3, "three")).unzip
+    *    // xs == ($Coll(1, 2, 3),
+    *    //        $Coll(one, two, three))
+    *    }}}
+    *
+    *  @tparam A1    the type of the first half of the element pairs
+    *  @tparam A2    the type of the second half of the element pairs
+    *  @param asPair an implicit conversion which asserts that the element type
+    *                of this $coll is a pair.
+    *  @return       a pair of ${coll}s, containing the first, respectively second
+    *                half of each element pair of this $coll.
+    */
+  def unzip[A1, A2](implicit asPair: A <:< (A1, A2)): (CC[A1], CC[A2]) = {
+    val unzipped = View.Unzip(coll)
+    (fromIterable(unzipped.left), fromIterable(unzipped.right))
```

Comment:
Not quite relevant here but I didn't catch it before: `left` and `right` are unfortunate names given their meaning in `Either`.